test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/initialgainorloss/InitialGainOrLossControllerSpec.scala

```diff
@@ -124,35 +130,85 @@ class InitialGainOrLossControllerSpec
+          )
+          val capacitorFillingOutReturn = individualFillingOutReturn.copy(draftReturn =
+            draftReturn.copy(triageAnswers = getTriageAnswersWithCapacitor())
+          )
+
+          val testData = List(
+            ("", individualFillingOutReturn, false),
+            (".trust", trustFillingOutReturn, false),
+            (".personalRep", personalRepresentativeFillingOutReturn, false),
+            (".capacitor", capacitorFillingOutReturn, false),
+            (".agent", individualFillingOutReturn, true)
           )
+          testData.map(keyWithReturn => test(keyWithReturn._1, keyWithReturn._2, keyWithReturn._3))
         }
       }
 
+      def test(key: String, fillingOutReturn: FillingOutReturn, isAgent: Boolean) = {
+        val session =
+          if (isAgent) SessionData.empty.copy(userType = Some(UserType.Agent), journeyStatus = Some(fillingOutReturn))
+          else
+            SessionData.empty.copy(
+              journeyStatus = Some(fillingOutReturn)
+            )
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(
+            session
+          )
+        }
+        checkPageIsDisplayed(
+          performAction(),
+          messageFromMessageKey(
+            s"initialGainOrLoss.title$key"
+          ), { doc =>
+            doc.select(".govuk-caption-xl").html() should include("Initial gain or loss")
+            doc.select("#initialGainOrLoss").html() should include(
+              messageFromMessageKey(s"initialGainOrLoss.title$key")
+            )
+            doc.select("#initialGainOrLoss-0-content > div > label").text() should be("Initial gain amount")
+            doc.select("#initialGainOrLoss-1-content > div > label").text() should be("Initial loss amount")
+            doc.select("#back").attr("href")                                shouldBe routes.InitialGainOrLossController.checkYourAnswers().url
+            doc.select("#content > article > form > details > div> ol > li:nth-child(2)").text() should include(
+              messageFromMessageKey(s"initialGainOrLoss.details.li2$key")
+            )
+            doc.select("#content > article > form > details > div> p").text() should startWith(
+              messageFromMessageKey(s"initialGainOrLoss.details.olTitle$key")
+            )
+            doc.select("#initialGainOrLoss-form-hint").text() should be(messageFromMessageKey(s"initialGainOrLoss.helpText$key"))
+            doc.select("#content > article > form").attr("action") shouldBe routes.InitialGainOrLossController
+              .submitInitialGainOrLoss()
+              .url
+          }
+        )
+      }
     }
+    def generateTrustSubscribedDetails() =
+      sample[SubscribedDetails].copy(name = Left(sample[TrustName]))
+
+    def generateIndividualSubscribedDetails() =
+      sample[SubscribedDetails].copy(name = Right(sample[IndividualName]))
+
+    def getTriageAnswersWithPersonalRepresentative() =
```

Comment:
is there a reason these method are `get` but the ones above are `generate`?

Reply:
my bad, fixed.